File: ios-client/StripeApplication/StripeApplication/Data/Models/ExternalAccount.swift
import Foundation

struct ExternalAccount: Codable, Identifiable {
    let id: String
    let object: String
    let bankName: String?
    let last4: String
    let routingNumber: String?
    let currency: String
    let country: String
    let defaultForCurrency: Bool
    let status: String?

    enum CodingKeys: String, CodingKey {
        case id, object, last4, currency, country, status
        case bankName = "bank_name"
        case routingNumber = "routing_number"
        case defaultForCurrency = "default_for_currency"
    }
}

struct ExternalAccountsResponse: Codable {
    let externalAccounts: [ExternalAccount]

    enum CodingKeys: String, CodingKey {
        case externalAccounts = "external_accounts"
    }
}

struct CreateExternalAccountRequest: Codable {
    let token: String
}

File: android-client/app/src/main/java/com/example/stripedemo/data/models/ExternalAccount.kt
package com.example.stripedemo.data.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ExternalAccount(
    val id: String,
    @Json(name = "object")
    val objectType: String,
    @Json(name = "bank_name")
    val bankName: String?,
    val last4: String,
    @Json(name = "routing_number")
    val routingNumber: String?,
    val currency: String,
    val country: String,
    @Json(name = "default_for_currency")
    val defaultForCurrency: Boolean,
    val status: String?
)

@JsonClass(generateAdapter = true)
data class ExternalAccountsResponse(
    @Json(name = "external_accounts")
    val externalAccounts: List<ExternalAccount>
)

@JsonClass(generateAdapter = true)
data class CreateExternalAccountRequest(
    val token: String
)
